src/server/browserContext.ts

```diff
@@ -93,10 +93,6 @@ export abstract class BrowserContext extends SdkObject {
     }
     this._closedStatus = 'closed';
     this._downloads.clear();
-    for (const [id, video] of this._browser._idToVideo) {
```

Comment:
Why removing this? It does not hurt to cleanup, just in case something went wrong and we leaked stuff.

Reply:
Because this is the reason the test is failing. The video gets removed from the map and after that reportFinished fails.